content/gans_and_augmentation.qmd

```diff
@@ -4,9 +4,98 @@ toc-title: '![](../imgs/cover.jpg){width=240px}<br> <h3>Augmentation of image da
+- The generator and discriminator can get stuck in a state where the generator produces low-quality samples that are easily distinguishable from real data, while the discriminator becomes too good at distinguishing between real and fake samples. In this case, it will be very hard for the generator to improve its performance over time. This is often referred to as vanishing gradients. To avoid this, techniques like Wasserstein GANs (WGAN) have been proposed, which use a different loss function that can help stabilize training and prevent mode collapse.
+- They can be computationally expensive to train, especially when dealing with high-dimensional data such as images.
+
+### Variants of GANs
+
+There are many variants of GANs that have been proposed in the literature to address some of these challenges and improve their performance. Some examples include:
+
+- Deep Convolutional Generative Adversarial Networks (DCGAN) @radfordUnsupervisedRepresentationLearning2015, which use convolutional layers in both the generator and discriminator to generate high-quality images.
+- Wasserstein GAN (WGAN) @arjovskyWassersteinGenerativeAdversarialNetworks2017, which uses the Wasserstein distance as a loss function instead of the traditional cross-entropy loss to improve stability and convergence properties.
+- StyleGAN @karrasStyleBasedGeneratorArchitecture2019, which uses a novel architecture that allows for fine-grained control over the style and content of generated images. It also introduces a new technique called style mixing, which allows for the creation of new styles by combining existing ones.
+- BigGAN @brockLargeScaleGANTraining2018, which uses a large batch size and spectral normalization to improve stability and convergence properties.
+- Progressive Growing GAN (PGGAN) @karrasProgressiveGrowingGenerativeAdversarialNetworks2017, which gradually increases the resolution of generated images over time to improve quality and stability.
+- CycleGAN @zhuUnpairedImageToImageTranslation2017, which uses a cycle consistency loss to enable unsupervised image-to-image translation between two domains without the need for paired data.
+- StarGAN @choiStarGANDeepLearningBasedFaceAttributeManipulation2018, which enables unsupervised image-to-image translation between multiple domains by learning a single mapping function that can transform images from one domain to any other domain.
+
+
+::: {.callout-note}
+
+## 📝 Task 
+
+Train one yourself!
+
+- implement a simple GAN architecture in pytorch (you can use [this noteook on kaggle](https://www.kaggle.com/code/songseungwon/pytorch-gan-basic-tutorial-for-beginner)) and train it on the [MNIST dataset](https://huggingface.co/datasets/ylecun/mnist)
```

Comment:
have you tried this?
We had an issue with some laptops not running cuda so this might be difficult in-session.

Reply:
yeah i know. i just don't know what else to do. gans are an arse to train, it is completely unrealistic to do this in a short 10 minute session.

Reply:
Maybe run some pretrained ones in colab?
I do not have any better idea though